### ISSUE TITLE
[JENKINS-11962] Symlinking lastSuccessful build shouldn't fail with concurrent jobs

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#11382]
<p>I had three builds running at the same time, and two of them finished during the same second. This lead to the following message on one of them:</p>

<p><tt>ln -s builds/2011-12-01_21-35-22 /var/lib/jenkins/jobs/my_job/builds/../lastSuccessful failed: 17 File exists</tt></p>

<p>The job still succeeded, so it's not a big deal.</p>

<p>This seems like a race condition between rm-ing the old symlink and creating the new one. Maybe ln -sf would work better? I assume it does its operations atomically.</p>

**Attachments:**

- [ScreenShot.png](https://issues.jenkins.io/secure/attachment/27993/ScreenShot.png)
- [bugchugger_cleanup_1098516.txt](https://issues.jenkins.io/secure/attachment/27994/bugchugger_cleanup_1098516.txt)
- [bugchugger_cleanup_1098517.txt](https://issues.jenkins.io/secure/attachment/27995/bugchugger_cleanup_1098517.txt)
- [bugchugger_cleanup_1098518.txt](https://issues.jenkins.io/secure/attachment/27996/bugchugger_cleanup_1098518.txt)
- [error](https://issues.jenkins.io/secure/attachment/27997/error)

---
<details><summary><i>Originally reported by <a title="Jørgen Tjernø" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jorgenpt">jorgenpt</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-11962" target="_blank">Symlinking lastSuccessful build shouldn't fail with concurrent jobs</a></i></summary>
<i><ul>
<li><b>reported by</b>: jorgenpt
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2023-06-23T00:57:44+00:00
<li><b>votes</b>: 4
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-07
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Can this be reproduced in more recent versions of Jenkins (no older than 8-10 weeks or so)? If so, what OS and what version and vendor of Java are you using? Please include log excerpts, content of the /systemInfo URL in your comment. Also relevant:<br/>
<a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rdrabens">rdrabens</a>:</i>
<p>I have reproduced this issue in Jenkins 1.588 running on Red Hat Enterprise Linux Server release 5.5 (Tikanga).  It is using Java 1.6.0_21-b06 from Sun (Oracle).  I’m not sure my company would be happy if I publicly posted the entire contents of /systemInfo, but if there are specific sections of interest, I could possibly remove the sensitive data and post them.   </p>

<p>It seems that one job gets stuck trying (and ultimately failing) to create a symlink.  While it is stuck, other jobs get stuck too.  The instant the first job finishes, the other jobs finish too.  The job in this example runs very quickly and also gets stuck for a fairly short period of time, but we have seen the same behavior in other jobs that normally run on the order of many seconds to many minutes and they can get stuck for up to 20 minutes.</p>

<p>In the screen shot that I have attached, you can see that jobs 1098516, 1098517, 1098518 took longer than the others.  Job 1098516 failed to create the symlink.  I’m attaching the console output of the three jobs.  I’m also attaching part of the Jenkins error log.  There is an error in the error log, but it is difficult to tell if it is related or not.  It is a frequent error and in other examples of this issue, the error does not always occur immediately before the stuck jobs finish.  </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>I think the Java 1.6 implementation needs JNA to work or something similar. Maybe try running Jenkins on Java 7.</p>

<p>Also, it doesn't look like you need the builds to run in parallel, as they are all done within milliseconds. Disabling that would likely prevent issues like that.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rdrabens">rdrabens</a>:</i>
<p>Thanks for the suggestions Daniel.  We plan to move to Java 7 in the near future.  Possibly that will fix the issue.  Good point about the parallel runs being unnecessary.  However, the other job we have that runs for much longer (and stalls for much longer) does require parallel runs.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kenpoole">kenpoole</a>:</i>
<p>This just happened to us on jenkins 1.644 running from the "official"<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/help_16.png" height="16" width="16" align="absmiddle" alt="" border="0"/> docker image.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>:</i>
<p>Jenkins no longer creates symbolic links by default.  Closing</p>
